src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -1176,6 +1212,31 @@ private string ResolvePSPath(string path)
             return result;
         }
 
+        /// <summary>
+        /// Gets the next literal path.
+        /// Filtering to ones that exist for the filesystem.
+        /// </summary>
+        /// <param name="command">
+        /// The command to search for.
+        /// </param>
+        /// <param name="provider">The provider that the command was found in.</param>
+        /// <returns>
+        /// Full path to the command.
+        /// </returns>
+        private string GetNextLiteralPathThatExists(string command, out ProviderInfo provider)
+        {
+            string resolvedPath = _context.LocationGlobber.GetProviderPath(command, out provider);
+
+            if (provider.Name.Equals("FileSystem", StringComparison.OrdinalIgnoreCase)
```

Comment:
Can provider be null?
```suggestion
            if (provider?.Name.Equals("FileSystem", StringComparison.OrdinalIgnoreCase)
```

Reply:
If it can we have this bug all over the code, and this should probably be cleaned up in a separate PR.